addon-entwicklung.md

```diff
@@ -1 +1,169 @@
 # AddOn Entwicklung
+
+* [Minimale Erfordernisse](#anker-minimal)
+* [package.yml](#anker-package-yml)
+  * [Erforderliche Angabe](#anker-yaml-required)
+  * [Optionale Angabe](#anker-yaml-optional)
+* [Optionale Dateien](#anker-files)
+* [Weitere nützliche Dateien](#anker-useful-files)
+* [Optionale Ordner](#anker-folders)
+* [Weitere nützliche Ordner](#anker-useful-folders)
+* [Software](#anker-software)
+* [Fragen und Antworten](#anker-faq)
+  * [MarkDown](#anker-faq-markdown)
+  * [Yaml](#anker-faq-yaml)
+
+
+<a name="anker-minimal"></a>
+## Minimale Erfordernisse
+Jedes AddOn hat einen eigenen Ordner mit einem einzigartigem und möglichst beschreibendem Namen.  
```

Comment:
Wo liegt der Ordner? Wo kommt der Name her, wenn man ein neues AddOn entwickelt? Wo wird er registriert? Welche Schreibweisen sind erlaubt?

Reply:
Erstmal vielen Dank @DavidBruchmann, dass Du hier die Initiative ergriffen hast! Ich weiß, dass aber @gupi auch an dem Thema arbeitet, es wäre schade, wenn zwei parallel arbeiten. Hattest Du schon angefangen, @gupi?
Das Inhaltsverzeichnis liegt übrigens hier: https://github.com/redaxo/docs/blob/master/dokumentation.md
Am besten das Kapitel markieren, das man bearbeiten möchte und/oder uns (@dtpop, @skerbis oder mir) Bescheid sagen, dann passieren keine Dopplungen.
Ein paar wenige Hinweise zur Schreibweise stehen in der README, ein paar Hinweise zur Formatierung hier: https://github.com/redaxo/docs/blob/master/_vorlage.md

Im Grunde geht es also um die beiden Kapitel AddOns-Installation und AddOns-Entwicklung. (Das Inhaltsverzeichnis kann bei gewichtigen Gründen natürlich modifiziert werden).

Hierzu finde ich die Hinweise von @schuer schon mal super. Bei Fragen zu Doku-Kapiteln können wir uns an @gharlan wenden, ggf. auch @tbaddade oder @dergel, die können/müssen uns Infos geben.

Die Passage über Markdown finde ich hier nicht wesentlich, sie lenkt in dieser Ausführlichkeit m.E: eher ab. Ich würde nur kurz erwähnen, dass man Markdown verwenden kann bei Hilfsdateien. Wir können aber gern überlegen, ob wir ein eigenes Kapitel Markdown (plus Textile) hinzunehmen.

Generell würde ich ins AddOn-Kapitel auch ein kleines Tutorial einbetten und/oder viele Code-Beispiele. So könnte man beim Durchgehen der AddOn-Dateien und Verzeichnisse  nach und nach den Code entwickeln für ein kleines Beispiel-AddOn, das irgendwas Kleines macht.
Es gab auf der alten Website ein Tutorial "AddOn-Entwicklung in sieben Schritten", damit konnten die Leute immer viel anfangen. Bei Interesse suche ich das raus.

Jetzt ist aber erstmal die Frage, wie wir das Kapitel hier voranbringen ohne uns in die Quere zu kommen. @DavidBruchmann und @gupi, wer von Euch könnte sich vorstellen, was zu machen?
Es ist ein prima Anfang, aber es ist erst 5 oder 10%, was im Idealfall noch kommen sollte.

Reply:
Evtl. hilft es sich im #doku Channel in Slack mal auszutauschen, bevor hier weitergearbeitet wird.

Reply:
@polarpixel Danke für den langen Kommentar.  
Bezüglich MarkDown habe ich bereits eine neue Seite "Dokumentation erstellen" vorgeschlagen, ich denke das ist auch für manche Nicht-Entwickler interessant.  
Bezüglich "aber es ist erst 5 oder 10%" gebe ich Dir vollkommen recht, es ist wirklich erst ein ganz kleiner Teil. Prinzipiell wäre es nicht schlecht schon mal ein paar Unterkapitel zu definieren, dann kann man sich da besser aufteilen.  
Bezüglich @gupi räume ich durchaus ein, daß er da evtl. nicht nur schon was vorbereitet hat, sondern auch mehr Wissen hat, da meine Erfahrung mit Redaxo sehr kurz ist. Dennoch hat er mir gegenüber den Eindruck erweckt, daß er an einer Zusammenarbeit kein Interesse hat - auch wenn er es zwar deutlich aber nicht konkret verbalisiert hat.  
Die Hinweise von @schuer sind tatsächlich sehr brauchbar, aber ich kann diese Fragen ohne Recherche nicht beantworten, wahrscheinlich fällt das jemand Anders wesentlich einfacher.   
Bezüglich der Code-Beispiele hatte ich eine Codebox vorgschlagen, das würde die Lesbarkeit der Gesamtseite sicher erhöhen, da man auch längeren Kode übersichtlich limitiert darstellen kann, evtl. geht das nur per CSS.
Bezüglich weiterer Inhalte denke ich das es Überschneidung mit Vorhandenem gibt, evtl. sogar die Verlagerung in AddOn-Entwicklungsseite, ich denke da z.B. an Fragmente und Formulare.
Da müßte man sehen, ob man dann eine Verlagerung entscheidet oder zwei Kapitel zum gleichen Thema an unterschiedlichen Stellen hat.

Reply:
Hallo David. Nach Absprache im Doku-Team, haben wir uns entschieden den PR anzunehmen und erstmal selbst an dem Kapitel weiterzuarbeiten. Wir werden den Text noch mal reviewen und überarbeiten, so dass eine Basis für die weitere Dokumentation entsteht. Im Anschluss werden wir das Kapitel für die weitere Bearbeitung an Euch wieder freigeben. Vielen Dank schon mal für die Vorarbeit.

Reply:
Prima, freut mich.
Dann bin ich mal gespannt wie es wird und wann es soweit ist.